src/charm.py

```diff
@@ -344,6 +344,26 @@ def is_cluster_initialised(self) -> bool:
         """Returns whether the cluster is already initialised."""
         return "cluster_initialised" in self.app_peer_data
 
+    @property
+    def is_cluster_restoring_backup(self) -> bool:
+        """Returns whether the cluster is restoring a backup."""
+        return "restoring-backup" in self.app_peer_data
+
+    @property
+    def is_cluster_restoring_to_time(self) -> bool:
+        """Returns whether the cluster is restoring a backup to a specific time."""
+        return "restore-to-time" in self.app_peer_data
+
+    @property
+    def is_unit_departing(self) -> bool:
+        """Returns whether the unit is departing."""
+        return "departing" in self.unit_peer_data
+
+    @property
+    def is_unit_stopped(self) -> bool:
+        """Returns whether the unit is stopped."""
+        return "stopped" in self.unit_peer_data
+
```

Comment:
If we are factoring those out, we might as well check if the peer data is not None.